### DISCUSSION TITLE
Bulk operations return code and rollback strategy

### DISCUSSION THREAD

**Discussion by chaen** [DIRACGrid/diracx#97] [Internal design]
When doing a bulk operations when part of the action failed (for example when killing jobs, only 1 job out of 2 succeeded)
* what should the http return code be ?
* what should be the atomicity of the operation: rollback everything or accept that only part of the operation is successful ?

**Comment by simon-mazenoux** [comment]
For me, given the nature of a REST-like interface, in case a job is no longer present in the DB, return an HTTP 404 code, providing details regarding which jobs were not found in the db, and rollback the transaction. This way, a user that still wants to kill his jobs might do it simply by removing the other jobs he has.

On the other hand, if we wanted to accept partly the operation, we would have to return a successful HTTP code (something like 200), and specify that some job ids were skipped in the return message. A user would then have no way to "unkill" his jobs if he wishes in case of an error.

**Reply by fstagni** [reply]
It feels more right, and maybe easier to handle, to rollback everything.

**Comment by chaen** [comment]
After discussion with @simon-mazenoux and @chrisburr we agreed that we should rollback everything
